Yank/pipeline.py

```diff
@@ -338,6 +339,12 @@ def prepare_phase(positions_file_path, parameters_file_path, ligand_dsl, system_
             logger.info("gro: %s" % positions_file_path)
 
         positions_file = openmm.app.GromacsGroFile(positions_file_path)
+        if ('nonbonded_method' in system_options) and (system_options['nonbonded_method'] in _NONPERIODIC_NONBONDED_METHODS):
+            # gro files must contain box vectors, so we must determine whether system is non-periodic or not from provided nonbonded options
+            # WARNING: This uses the private API for GromacsGroFile, and may break.
+            logger.info('nonbonded_method = %s, so removing periodic box vectors from gro file' % system_options['nonbonded_method'])
+            for (frame, box_vectors) in enumerate(positions_file._periodicBoxVectors):
```

Comment:
There is no `setPeriodicBoxVectors()` in the public API, so I would have to use the private API in the very next line anyway. I'll make a note that we should add this.